### DISCUSSION TITLE
How to ensure high availability in a multi-datacenter deployment?

### DISCUSSION THREAD

**Discussion by wushengwusheng** [ccfos/nightingale#1251] [Q&A] [answered]
多机房的话各自往各自的n9e server写，假设一个机房500台服务器，需要多少个n9e server呀？或者说如何保证多服务器往n9e server写高可用负载均衡？

**Comment by UlricQin** [comment, marked as answer]
官网文档说的很清楚了，而且有架构图，拜托动动脑子：

<img width="680" alt="image" src="https://user-images.githubusercontent.com/2769466/200110110-4610482e-6ed7-4f8a-a1a1-4ac1ca13df2d.png">

agent(可能是categraf、telegraf、Grafana-agent、datadog-agent)往LB写，LB后面是多个n9e-server，这多个n9e-server使用相同的ClusterName，组成一个集群，如果某个 n9e-server 挂了，LB会自动踢掉。

> 假设一个机房500台服务器，需要多少个n9e server呀？

没啥前提条件，让人怎么回答，，，n9e-server部署在1c1g的机器上，跟部署在128c256g的机器上性能能一样么？！千兆网卡、万兆网卡能一样么？！

这个问题完全可以测出来，先部署100台agent看看n9e-server的机器的各项性能指标，如果很宽松，继续部署不就完事了，每个环境不同，自己的环境的测试结果显然是最靠谱可信的！

**Reply by wushengwusheng** [reply]
想不通 有最后一句话不就行了吗～～在你没回答之前我就是这么做的 无非就是想有个答案确认下，这就开始动不动脑子了吗？你们部署服务会在1c1g的服务器上吗，你脑子呢？

**Reply by wushengwusheng** [reply]
我一共问俩问题 一个是说要付费 一个是动不动脑子 您可真刑

**Reply by UlricQin** [reply]
这图这么明显，如果动脑子，会不知道？问问题没有前提，别人怎么回答？远程排查我是要花费时间的，你又不是我的甲方， 我收点钱怎么了，我这是付出我的时间，你遇到问题我凭什么免费给你远程解决，我欠你的？

**Reply by wushengwusheng** [reply]
是是是 您说的对

**Reply by wushengwusheng** [reply]
我要去高铁站检票口，因为第一次来虽然有个大致的方向，但还是选择和站点工作人员确认下是不是这么走。结果，站点工作人员说：“摆脱你睁开眼睛看看，上方都写着的；如果你付我钱我可以告诉你”
@UlricQin 希望您以后也能被这么对待

**Reply by UlricQin** [reply]
> 我要去高铁站检票口，因为第一次来虽然有个大致的方向，但还是选择和站点工作人员确认下是不是这么走。结果，站点工作人员说：“摆脱你睁开眼睛看看，上方都写着的；如果你付我钱我可以告诉你” @UlricQin 希望您以后也能被这么对待

高铁站工作人员是服务人员，是拿工资提供服务的人员，可我不是啊，你和我之间没有这个关系啊。。。

**Reply by wushengwusheng** [reply]
是啊 问问题的同学们 你应该让他们直接看代码

**Reply by UlricQin** [reply]
> 是啊 问问题的同学们 你应该让他们直接看代码

你说对了，用开源软件看代码一点都不过分。开源软件，是一堆志同道合的人一起搞一个软件，我们作为创始团队，只是搭个架子，大家一起贡献打造产品，然后解决痛点问题大家一起受益。开源软件有基本的文档，代码放出来了，就足够了。不是甲乙方的关系。

**Reply by wushengwusheng** [reply]
有人问问题有人回答这就是甲乙方的关系吗？我也没把您当甲方，我只是问了个不太确认的问题，就被您要求多动动脑了。我一共问俩问题，第一个问题上来就是要求要付费，第二个问题被要求多动脑，我感觉不到您的尊重在哪。

**Reply by UlricQin** [reply]
> 有人问问题有人回答这就是甲乙方的关系吗？我也没把您当甲方，我只是问了个不太确认的问题，就被您要求多动动脑了。我一共问俩问题，第一个问题上来就是要求要付费，第二个问题被要求多动脑，我感觉不到您的尊重在哪。

天地良心，我有没有给你先发文档？我是看社区里的人跟你讨论了半天，没有思路，我也没有思路才建议远程解决的呀。我们就到这吧，讨论这个没有意义，核心冲突是大家对开源协作模式的认知不同。
